src/common/inference/bm25.rs

```diff
@@ -0,0 +1,163 @@
+use std::borrow::Cow;
+use std::collections::{BTreeMap, HashMap};
+use std::sync::Arc;
+
+use collection::operations::point_ops::VectorPersisted;
+use itertools::Itertools;
+use murmur3::murmur3_32_of_slice;
+use segment::data_types::index::{StemmingAlgorithm, StopwordsInterface, TokenizerType};
+use segment::index::field_index::full_text_index::stop_words::StopwordsFilter;
+use segment::index::field_index::full_text_index::tokenizers::{
+    Stemmer, Tokenizer, TokenizerConfig,
+};
+use serde::{Deserialize, Serialize};
+
+/// Configuration of the local bm25 models.
+#[derive(Debug, Deserialize, Serialize, Clone)]
+pub struct Bm25Config {
+    #[serde(default = "default_k")]
+    pub k: f64,
+    #[serde(default = "default_b")]
+    pub b: f64,
+    #[serde(default = "default_avg_len")]
+    pub avg_len: f64,
+    #[serde(default)]
+    pub tokenizer: TokenizerType,
+    #[serde(default)]
+    pub tokenizer_config: Option<InferenceTokenizerConfig>,
```

Comment:
I don't think it should be called `tokenizer_config`, in fact, I don't think it should be called anything. I would prefer to use `#[serde(flatten)]` here.